hypervisor/arch/x86/cpu.c

```diff
@@ -227,6 +227,17 @@ static int32_t hardware_detect_support(void)
 		return -ENODEV;
 	}
 
+	if (!cpu_has_cap(X86_FEATURE_ERMS)) {
+		pr_fatal("%s, enhanced rep stosb/movsb not supported\n", __func__);
+		return -ENODEV;
+	} else {
+		uint32_t misc_enable = (uint32_t)msr_read(MSR_IA32_MISC_ENABLE);
+		if ((misc_enable & MSR_IA32_MISC_ENABLE_FAST_STRING) == 0U) {
+			pr_fatal("%s, fast string is not enabled\n", __func__);
```

Comment:
In this case, we should enable it, not?

Reply:
And this seems need to be per-AP enable...

Reply:
"Fast-Strings EnableWhen set, the fast-strings feature (for REPMOVS and REP STORS) is enabled (default);when clear, fast-strings are disabled."
We don't need to enable fast string by manual (ACRN will not disable it). But I think I have make a mistake. I should first check this bit then check X86_FEATURE_ERMS.
Will update it.